fem/gridfunc.cpp

```diff
@@ -4550,4 +4550,409 @@ GridFunction *Extrude1DGridFunction(Mesh *mesh, Mesh *mesh2d,
    return sol2d;
 }
 
+KDTreeNodalProjection::KDTreeNodalProjection(GridFunction& dest_)
+{
+   dest=&dest_;
+   FiniteElementSpace* space=dest->FESpace();
+   Mesh* mesh=space->GetMesh();
+   int dim=mesh->SpaceDimension();
+
```

Comment:
Can we add an assertion to ensure that the fe space is really a nodal fe space?

Reply:
I am not sure how can we do this.

Reply:
One solution:

```
MFEM_VERIFY(
    dynamic_cast<const H1_FECollection*>(fes.FEColl()) != nullptr || 
    dynamic_cast<const L2_FECollection*>(fes.FEColl()) != nullptr, 
    "Error!");
```

Reply:
Is this really a solution? I.e. do we have the guarantee that every finite element formulation in H1 and L2 is always  a nodal formulation?

Reply:
Good point, I was too quick and didn't think this through. I don't have the full expertise to come up with a good solution here.

Reply:
I think the above is good - I will add it - users should read the documentation :-)